src/spectrum.jl

```diff
@@ -0,0 +1,177 @@
+export spectrum, spectrum_correlation_fft
+export SpectrumSolver, ExponentialSeries, PseudoInverse
+
+abstract type SpectrumSolver end
+
+@doc raw"""
+    ExponentialSeries(; tol = 1e-14, calc_steadystate = false)
+
+A solver which solves [`spectrum`](@ref) by finding the eigen decomposition of the Liouvillian [`SuperOperator`](@ref) and calculate the exponential series.
+"""
+struct ExponentialSeries{T<:Real,CALC_SS} <: SpectrumSolver
+    tol::T
+    ExponentialSeries(tol::T, calc_steadystate::Bool = false) where {T} = new{T,calc_steadystate}(tol)
+end
+
+ExponentialSeries(; tol = 1e-14, calc_steadystate = false) = ExponentialSeries(tol, calc_steadystate)
+
+@doc raw"""
+    PseudoInverse(; alg::SciMLLinearSolveAlgorithm = KrylovJL_GMRES())
+
+A solver which solves [`spectrum`](@ref) by finding the inverse of Liouvillian [`SuperOperator`](@ref) using the `alg`orithms given in [`LinearSolve.jl`](https://docs.sciml.ai/LinearSolve/stable/).
+"""
+struct PseudoInverse{MT<:SciMLLinearSolveAlgorithm} <: SpectrumSolver
+    alg::MT
+end
+
+PseudoInverse(; alg::SciMLLinearSolveAlgorithm = KrylovJL_GMRES()) = PseudoInverse(alg)
+
+@doc raw"""
+    spectrum(H::QuantumObject,
+        ωlist::AbstractVector,
+        c_ops::Union{Nothing,AbstractVector,Tuple},
+        A::QuantumObject{<:AbstractArray{T2},OperatorQuantumObject},
+        B::QuantumObject{<:AbstractArray{T3},OperatorQuantumObject};
+        solver::SpectrumSolver=ExponentialSeries(),
+        kwargs...)
+
+Calculate the spectrum of the correlation function
+
+```math
+S(\omega) = \int_{-\infty}^\infty \lim_{t \rightarrow \infty} \left\langle \hat{A}(t + \tau) \hat{B}(t) \right\rangle e^{-i \omega \tau} d \tau
+```
+
+See also the following list for `SpectrumSolver` docstrings:
+- [`ExponentialSeries`](@ref)
+- [`PseudoInverse`](@ref)
+"""
+function spectrum(
+    H::QuantumObject{MT1,HOpType},
+    ωlist::AbstractVector,
+    c_ops::Union{Nothing,AbstractVector,Tuple},
```

Comment:
Here instead, we need the collapse operators, I think. Because the spectrum needs some dissipation I would say.

Reply:
But we should still support `c_ops::Nothing`, cause users might directly specify `H` as Liouvillian.

Reply:
Yeye